src/theory/quantifiers/bv_inverter.cpp

```diff
@@ -261,17 +262,112 @@ Node BvInverter::getPathToPv(Node lit, Node pv, Node sv, Node pvs,
+                              Node lit,
+                              bool pol,
+                              std::vector<unsigned>& path,
+                              BvInverterModelQuery* m,
+                              BvInverterStatus& status) {
+
   unsigned index;
   unsigned nchildren;
   NodeManager* nm = NodeManager::currentNM();
+  Kind k;
+
+  do {
+    Assert (!path.empty());
+    index = path.back();
+    Assert(index < lit.getNumChildren());
+    path.pop_back();
+    k = lit.getKind();
+    Trace("bv-invert") << "####### kind " << k << std::endl;
+    if (k != NOT) break;
+    pol = !pol;
+    lit = lit[index];
+  } while (k == NOT);
+
+  Assert (k == EQUAL
+       || k == BITVECTOR_ULT
+       || k == BITVECTOR_ULTBV
+       || k == BITVECTOR_SLT
+       || k == BITVECTOR_SLTBV);
+
+  Assert (k != EQUAL || pol == true);
+
+    Trace("bv-invert") << "####### kind " << k << std::endl;
+  Node sv_t = lit[index];
+  Node t = lit[1-index];
+
+  if (k == BITVECTOR_ULT || k == BITVECTOR_ULTBV) {
+    Trace("bv-invert") << "#######################################" << std::endl;
+    TypeNode solve_tn = sv_t.getType();
+    Node x = getSolveVariable(solve_tn);
+    Node sc, scr, scl;
+    if (index == 0) {
+      scr = nm->mkNode(k, x, t);
+      if (pol == true) {
+        /* x < t
+         * with side condition:
+         * t != 0  */
+        scl = nm->mkNode(DISTINCT, t, bv::utils::mkZero(bv::utils::getSize(t)));
+        sc = nm->mkNode(IMPLIES, scl, scr);
+      } else {
+        Trace("bv-invert") << "pol false 1" << std::endl;
+        sc = scr;
+      }
+    } else if (index == 1) {
+      scr = nm->mkNode(k, t, x);
+      if (pol == true) {
+        /* t < x
+         * with side condition:
+         * t != 1...1  */
+        scl = nm->mkNode(DISTINCT, t, bv::utils::mkOnes(bv::utils::getSize(t)));
+        sc = nm->mkNode(IMPLIES, scl, scr);
+      } else {
+        Trace("bv-invert") << "pol false 2" << std::endl;
+        sc = scr;
+      }
+    }
+    status.d_conds.push_back(sc);
+    /* t = skv (fresh skolem constant)  */
+    Node skv = getInversionNode(sc, solve_tn);
+    t = skv;
+  } else if (k == BITVECTOR_SLT || k == BITVECTOR_SLTBV) {
+    TypeNode solve_tn = sv_t.getType();
+    Node x = getSolveVariable(solve_tn);
+    Node sc, scr, scl;
+    unsigned w = bv::utils::getSize(t);
+    if (index == 0) {
+      scr = nm->mkNode(k, x, t);
+      if (pol == true) {
+        /* x < t
+         * with side condition:
+         * t != 10...0 */
+        Node max = bv::utils::mkConst(BitVector(w).setBit(w - 1));
+        scl = nm->mkNode(DISTINCT, max, t);
+        sc = nm->mkNode(IMPLIES, scl, scr);
+      } else {
+        sc = scr;
+      }
+    } else if (index == 1) {
+      scr = nm->mkNode(k, t, x);
+      if (pol == true) {
+        /* t < x
+         * with side condition:
+         * t != 01...1  */
+        BitVector bv = BitVector(w).setBit(w - 1);
+        Node min = bv::utils::mkConst(~bv);
```

Comment:
This variable should be named `max` then.

Reply:
Ditto, but lower bound.

Reply:
Ditto, see above.